core/src/main/scala/org/apache/spark/h2o/H2OContext.scala

```diff
@@ -425,9 +422,17 @@ object H2OContext extends Logging {
     }
 
     override protected def getSparklingWaterHeartBeatEvent(): SparklingWaterHeartbeatEvent = {
-      val cloudV3 = getCloudInfo()
-      val memoryInfo = cloudV3.nodes.map(node => (node.ip_port, PrettyPrint.bytes(node.free_mem)))
-      SparklingWaterHeartbeatEvent(cloudV3.cloud_healthy, System.currentTimeMillis(), memoryInfo)
+      try {
```

Comment:
This is executed in thread which is responsible for checking the hearbeats. Right now the name is a bit confusing, I would like to refactor the thread name to something more explicit like H2OClusterHeartBeatThread but wanted to keep this change as small as possible, will do in next PR

Reply:
After the client gets separated, will the heartbeat thread still exist?

Reply:
Yup, this heartbeat is checking the worker nodes, not the client. Right now it is used to fill the UI with info obtained from worker nodes & in this separated situation also for stopping the H2OContext if it's not healthy. I think it just deserves a better name.

Does it make sense?

Reply:
Ok thanks for the explanation!